download/oci_download_test.go

```diff
@@ -382,6 +383,43 @@ func TestOCICustomAuthPlugin(t *testing.T) {
 	}
 }
 
+func TestOCIRawDataHasNonZeroSize(t *testing.T) {
+	var buf bytes.Buffer
+	ctx := context.Background()
+	fixture := newTestFixture(t)
+	fixture.server.expEtag = "sha256:c5834dbce332cabe6ae68a364de171a50bf5b08024c27d7c08cc72878b4df7ff"
+
+	updates := make(chan *Update)
+
+	config := Config{}
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+
+	d := NewOCI(config, fixture.client, "ghcr.io/org/repo:latest", t.TempDir()).WithCallback(func(_ context.Context, u Update) {
+		updates <- &u
+	}).WithBundlePersistence(true)
+
+	d.Start(ctx)
+
+	// Give time for some download events to occur
+	time.Sleep(1 * time.Second)
+
+	u1 := <-updates
+
+	buf.ReadFrom(u1.Raw)
+
+	if u1.Raw == nil {
+		t.Fatal("expected bundle reader to be non-nil")
+	}
+
+	if buf.Len() == 0 {
```

Comment:
We should probably check `u1.Size == 0` here. Also for completeness can we do something like [this](https://github.com/open-policy-agent/opa/blob/main/download/download_test.go#L102) test.

Reply:
I see, looks like another bug - the Size is not being filled in the `downloaderResponse` then, since it's not there [now](https://github.com/open-policy-agent/opa/blob/a94d87f52945d7fdc89c940dfa373af21d840c68/download/oci_download.go#L201).
Let me fix that

Reply:
@ashutosh-narkar What do you think if I move [count](https://github.com/open-policy-agent/opa/blob/a94d87f52945d7fdc89c940dfa373af21d840c68/download/download.go#L419) struct from download package to `util` for reusing it in `oci_download`?

Reply:
I checked the code in the `oci_download` and `download` files, and `oci_download` has several dependencies with the `download` package, `Update` struct as an example, not sure how best to proceed here. 
Just leave `count` in `download` then?

Reply:
Pushed changes